src/utils/propertyTreeHelpers.js

```diff
@@ -112,23 +112,56 @@ export function removeLastWordFromUri(uri) {
 // Returns whether a property should be visible in the gui
 export function isPropertyVisible(properties, uri) {
   const property = properties[uri];
+  const visibility = properties['OpenSpaceEngine.PropertyVisibility'];
 
   const splitUri = uri.split('.');
   if (splitUri.length > 1) {
     if (splitUri[splitUri.length - 1] === 'Enabled')
       return false;
   }
 
-  return property &&
-         property.description &&
-         property.description.MetaData &&
-         property.description.MetaData.Visibility !== 'Hidden';
+  if (property && property.description && property.description.MetaData &&
+      property.description.MetaData.Visibility) {
+    let propertyVisibility = "";
+    switch(property.description.MetaData.Visibility) {
+      case 'Hidden':
+        propertyVisibility = 5;
+        break;
+      case 'Developer':
+        propertyVisibility = 4;
+        break;
+      case 'AdvancedUser':
+        propertyVisibility = 3;
+        break;
+      case 'User':
+        propertyVisibility = 2;
+        break;
+      case 'NoviceUser':
+        propertyVisibility = 1;
+        break;
+      case 'Always':
+        propertyVisibility = 0;
+        break;
+      default:
+        propertyVisibility = 0;
+        break;
+    }
+
+    return visibility.value >= propertyVisibility;
+  }
+  else {
```

Comment:
I suggest flipping this statement and doing an "early out" return for the false case. So that the big if-statement isn't needed. It reduces the indentation and often increases readability

I.e.:
```js
if (!(property && property.description && property.description.MetaData &&
    property.description.MetaData.Visibility)) 
{
  return false;
}

// ... here comes the switch!
```